core/src/main/java/org/apache/carbondata/scan/executor/QueryExecutorFactory.java

```diff
@@ -18,15 +18,69 @@
  */
 package org.apache.carbondata.scan.executor;
 
+import java.util.List;
+
+import org.apache.carbondata.core.carbon.metadata.datatype.DataType;
+import org.apache.carbondata.core.carbon.metadata.encoder.Encoding;
 import org.apache.carbondata.scan.executor.impl.DetailQueryExecutor;
+import org.apache.carbondata.scan.executor.impl.VectorDetailQueryExecutor;
+import org.apache.carbondata.scan.model.QueryDimension;
+import org.apache.carbondata.scan.model.QueryMeasure;
+import org.apache.carbondata.scan.model.QueryModel;
+import org.apache.carbondata.scan.result.vector.CarbonColumnVector;
+import org.apache.carbondata.scan.result.vector.CarbonColumnarBatch;
+import org.apache.carbondata.scan.result.vector.impl.CarbonColumnVectorImpl;
 
 /**
  * Factory class to get the query executor from RDD
  * This will return the executor based on query type
  */
 public class QueryExecutorFactory {
 
-  public static QueryExecutor getQueryExecutor() {
-    return new DetailQueryExecutor();
+  public static QueryExecutor getQueryExecutor(QueryModel queryModel) {
+    if (queryModel.isVectorReader()) {
+      return new VectorDetailQueryExecutor();
+    } else {
+      return new DetailQueryExecutor();
+    }
+  }
+
+  public static CarbonColumnarBatch createColuminarBatch(QueryModel queryModel) {
+    int batchSize = 10000;
+    List<QueryDimension> queryDimension = queryModel.getQueryDimension();
+    List<QueryMeasure> queryMeasures = queryModel.getQueryMeasures();
+    CarbonColumnVector[] vectors =
+        new CarbonColumnVector[queryDimension.size() + queryMeasures.size()];
+    for (int i = 0; i < queryDimension.size(); i++) {
+      QueryDimension dim = queryDimension.get(i);
+      if (dim.getDimension().hasEncoding(Encoding.DIRECT_DICTIONARY)) {
+        vectors[dim.getQueryOrder()] = new CarbonColumnVectorImpl(batchSize, DataType.LONG);
+      } else if (!dim.getDimension().hasEncoding(Encoding.DICTIONARY)) {
+        vectors[dim.getQueryOrder()] =
+            new CarbonColumnVectorImpl(batchSize, dim.getDimension().getDataType());
+      } else if (dim.getDimension().isComplex()) {
+        vectors[dim.getQueryOrder()] = new CarbonColumnVectorImpl(batchSize, DataType.STRUCT);
+      } else {
+        vectors[dim.getQueryOrder()] = new CarbonColumnVectorImpl(batchSize, DataType.INT);
+      }
+    }
+
+    for (int i = 0; i < queryMeasures.size(); i++) {
+      QueryMeasure msr = queryMeasures.get(i);
+      switch (msr.getMeasure().getDataType()) {
+        case SHORT:
+        case INT:
+        case LONG:
+          vectors[msr.getQueryOrder()] =
+              new CarbonColumnVectorImpl(batchSize, msr.getMeasure().getDataType());
+          break;
+        case DECIMAL:
+          vectors[msr.getQueryOrder()] = new CarbonColumnVectorImpl(batchSize, DataType.DECIMAL);
+          break;
+        default:
+          vectors[msr.getQueryOrder()] = new CarbonColumnVectorImpl(batchSize, DataType.DOUBLE);
+      }
```

Comment:
No we can't use as we support few datatypes while storing reading measure data.
Anyway this method is not used now, I am removing it